resources/js/views/settings/roomTypes/View.vue

```diff
@@ -0,0 +1,250 @@
+        <b-row class='my-1 float-right'>
+          <b-col sm='12'>
+            <b-button
+              :disabled='isBusy'
+              variant='secondary'
+              @click="$router.push({ name: 'settings.room_types' })">
+              <i class='fas fa-arrow-left'></i> {{ $t('app.back') }}
+            </b-button>
+            <b-button
+              :disabled='isBusy'
+              variant='success'
+              type='submit'
+              class='ml-1'
+              v-if='!viewOnly'>
+              <i class='fas fa-save'></i> {{ $t('app.save') }}
+            </b-button>
+          </b-col>
+        </b-row>
+      </b-container>
+    </b-form>
+
+    <b-modal
+      :static='modalStatic'
+      :busy='isBusy'
+      ok-variant='danger'
+      cancel-variant='dark'
+      @ok='forceOverwrite'
+      @cancel='refreshRoomType'
+      :hide-header-close='true'
+      :no-close-on-backdrop='true'
+      :no-close-on-esc='true'
+      ref='stale-roomType-modal'
+      :hide-header='true'>
+      <template v-slot:default>
+        <h5>{{ staleError.message }}</h5>
+      </template>
+      <template v-slot:modal-ok>
+        <b-spinner small v-if="isBusy"></b-spinner>  {{ $t('app.overwrite') }}
+      </template>
+      <template v-slot:modal-cancel>
+        <b-spinner small v-if="isBusy"></b-spinner>  {{ $t('app.reload') }}
+      </template>
+    </b-modal>
+  </b-overlay>
+</template>
+
+<script>
+import Base from '../../../api/base';
+import FieldErrors from '../../../mixins/FieldErrors';
+import { mapGetters } from 'vuex';
+import env from '../../../env';
+import VSwatches from 'vue-swatches';
+import 'vue-swatches/dist/vue-swatches.css';
+
+export default {
+  mixins: [FieldErrors],
+  components: {
+    VSwatches
+  },
+  props: {
+    id: {
+      type: [String, Number],
+      required: true
+    },
+
+    viewOnly: {
+      type: Boolean,
+      required: true
+    },
+
+    modalStatic: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      settings: 'session/settings'
+    })
+
+  },
+
+  data () {
+    return {
+      loaded: false,
+      isBusy: false,
+      errors: {},
+      staleError: {},
+      model: {
+        description: null,
+        short: null,
+        color: '#4a5c66'
+      },
+      swatches: ['#4a5c66', '#80ba24', '#9C132E', '#F4AA00', '#00B8E4', '#002878']
+    };
+  },
+
+  /**
+   * Loads the role from the backend and also a part of permissions that can be selected.
+   */
+  mounted () {
+    if (this.id !== 'new') {
+      this.loadRoomType();
+    } else { this.loaded = true; }
+  },
+
+  methods: {
+
+    /**
+     * Load the room type from the server api
+     *
+     */
+    loadRoomType () {
+      this.isBusy = true;
+
+      Base.call(`roomTypes/${this.id}`).then(response => {
+        this.model = response.data.data;
+        this.loaded = true;
+      }).catch(response => {
+        Base.error(response, this.$root, response.message);
```

Comment:
We should return here to the index page if the model wasn't found!

Reply:
I will also display a 404 message. The serverside message isn't great, so i created https://github.com/THM-Health/PILOS/issues/86